### DISCUSSION TITLE
Running object-detector-python example behind a proxy

### DISCUSSION THREAD

**Discussion by igor-rud-kr** [AxisCommunications/acap-computer-vision-sdk-examples#35] [Q&A] [answered]
I’m trying to run object detection example (https://github.com/AxisCommunications/acap-computer-vision-sdk-examples/tree/master/object-detector-python) on a P3265-LV.
Both the server that I am building the example and the camera are behind a proxy. I can configure proxy settings for the docker on the server, and can execute all the steps successfully up to this one:

docker-compose --tlsverify -H tcp://$AXIS_TARGET_IP:$DOCKER_PORT --env-file ./config/env.$ARCH.$CHIP up

it displays the following message for 15 seconds
[+] Running 0/1
⠹ inference-server Pulling

and then displays the following error message:
⠿ inference-server Error                                                                                                                                                              15.0s
Error response from daemon: Get https://registry-1.docker.io/v2/: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)

This message seem to come from the docker inside the camera, not from the build server. When I ssh to the camera and execute
systemctl status sdkdockerdwrapper.service -l
the output shows that the docker service inside the camera wants to connect to the registry:

time="2022-06-03T15:28:08.920695739-04:00" level=warning msg="Error getting v2 registry: Get \https://registry-1.docker.io/v2/\: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)"
time="2022-06-03T15:28:08.920840620-04:00" level=info msg="Attempting next endpoint for pull after error: Get \https://registry-1.docker.io/v2/\: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)"
time="2022-06-03T15:28:08.926380155-04:00" level=error msg="Handler for POST /v1.41/images/create returned error: Get \https://registry-1.docker.io/v2/\: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)"
…
time="2022-06-03T15:55:46.045734742-04:00" level=warning msg="Error getting v2 registry: Get \https://registry-1.docker.io/v2/\: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)"
time="2022-06-03T15:55:46.045907463-04:00" level=info msg="Attempting next endpoint for pull after error: Get \https://registry-1.docker.io/v2/\: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)"
time="2022-06-03T15:55:46.050668516-04:00" level=error msg="Handler for POST /v1.41/images/create returned error: Get \https://registry-1.docker.io/v2/\: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)"

Is there a workaround for this issue? Thank you!

**Comment by mattias-kindborg-at-work** [comment]
Hi @igor-rud-kr and welcome!

I'll move this discussion into https://github.com/AxisCommunications/acap-computer-vision-sdk-examples. The maintainers will hopefully be able to help you on your way.

**Comment by Corallo** [comment, marked as answer]
Hello Igor

If docker can't pull the image from the camera, a quick fix is to upload the runtime-image manually like you do for the other two images:
docker save axisecp/acap-runtime:0.6-aarch64 | docker --tlsverify  -H tcp://$AXIS_TARGET_IP:$DOCKER_PORT load

**Comment by igor-rud-kr** [comment]
Thank you! 

The error message that I am getting now is 

object-detector-python-object-detector-python-1  | standard_init_linux.go:219: exec user process caused: exec format error
object-detector-python-acap_dl-models-1          | standard_init_linux.go:219: exec user process caused: exec format error
object-detector-python-object-detector-python-1 exited with code 1
object-detector-python-acap_dl-models-1 exited with code 1
object-detector-python-inference-server-1        | Server listening on 0.0.0.0:8501

Can you please point me in the right direction - how can I trace this issue?

**Reply by Corallo** [reply]
This seems an architecture missmatch, P3265-LV is an aarch64 camera, did you export the [right architecture](https://github.com/AxisCommunications/acap-computer-vision-sdk-examples/tree/master/object-detector-python#export-the-environment-variable-for-the-architecture)?

**Reply by igor-rud-kr** [reply]
Yes, this was the issue, thank you!
